x/mint/keeper/hooks.go

```diff
@@ -14,18 +14,20 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epoch epochTypes.Epoch) {
 
 func (k Keeper) AfterEpochEnd(ctx sdk.Context, epoch epochTypes.Epoch) {
 	params := k.GetParams(ctx)
-	// Check if we have hit an epoch where we update the inflation parameter.
-	// Since epochs only update based on BFT time data, it is safe to store the "halvening period time"
-	// in terms of the number of epochs that have transpired
-	// If it's not time to mint coins, exit
-	if int64(epoch.GetCurrentEpoch()) < k.GetLastHalvenEpochNum(ctx)+params.ReductionPeriodInEpochs {
+	minter := k.GetMinter(ctx)
+
+	// If the current yyyy-mm-dd of the epoch timestamp matches any of the scheduled token release then proceed to mint
+	lastRelaseDate := k.GetLastTokenReleaseDate(ctx)
+	scheduledTokenRelease := types.GetScheduledTokenRelease(epoch, lastRelaseDate, params.GetTokenReleaseSchedule())
+	if scheduledTokenRelease == nil {
+		ctx.Logger().Debug(fmt.Sprintf("No release at epoch time %s; last release %s", epoch.GetCurrentEpochStartTime().String(), lastRelaseDate.Format(types.TokenReleaseDateFormat)))
 		return
 	}
-	// Halven the reward per halven period
-	minter := k.GetMinter(ctx)
-	minter.EpochProvisions = minter.NextEpochProvisions(params)
+
+	minter.EpochProvisions = sdk.NewDec(scheduledTokenRelease.GetTokenReleaseAmount())
```

Comment:
Still keeping `EpochProvisions` around as a way to tell the amount of tokens that was last minted